agent/internal/config/jwt.go

```diff
@@ -0,0 +1,36 @@
+package config
+
+import (
+	"fmt"
+	"github.com/form3tech-oss/jwt-go"
+)
+
+type ValidJWT struct {
+	Token string
+}
+
+func (field *ValidJWT) SetValue(unvalidatedToken string) error {
+	if unvalidatedToken == "" {
+		return fmt.Errorf("grpc token can not be unset")
+	}
+
+	token, err := ValidateAndCreateJWT(unvalidatedToken)
+	if err != nil {
+		return err
+	}
+
+	*field = token
+	return nil
+}
+
+func ValidateAndCreateJWT(unvalidatedToken string) (ValidJWT, error) {
+
+	token := ValidJWT{
+		Token: unvalidatedToken,
+	}
+
+	jwtParser := jwt.Parser{}
+	_, _, err := jwtParser.ParseUnverified(unvalidatedToken, &jwt.StandardClaims{})
```

Comment:
Here is an example token for local development. Add validation if the payload contains the all fields as in the token, every field is required (iat, iss, sub).

`eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2NjQ4NzQxMDM5NzQsImlzcyI6IjE3Mi4xNy4wLjE6NTAwMCIsInN1YiI6ImIzZmYwMmU5LWU4MDUtNDhjYy1hYzNkLWRhMDVhZTEwMjNkZiJ9.qVlU38JwsLFrRf85PMhMFjZ7Jm4wyxWovDCa7nPNV0g`

Reply:
`jwtParser.ParseUnverified` checks all the anatomy of the JSON token and in case of some missing values which should be required -- it returns a value which states that. Why to introduce additional checks?

Reply:
The first check is to tests that the construct, JWT is valid, if it is valid we need crucial parts to be defined in the payload, those have impact on agent behavior, if they are missing it is an invalid configuration, as stated, every field of those listed above are strictly required, therefore we should check for existence on startup.

Reply:
@trkohler payload validation missing. How we can help with this task to you?